stablehlo/tests/ops_stablehlo.mlir

```diff
@@ -1475,38 +1469,46 @@ func.func @clamp_compatible_dynamic_match_static(%arg0: tensor<?xi32>, %arg1: te
 
 // -----
 
-func.func @clamp_invalid_clamp_element_type(%arg0: tensor<1xi32>, %arg1: tensor<1xf32>) -> tensor<1xi32> {
-  // expected-error@+1 {{'stablehlo.clamp' op requires the same element type for all operands and results}}
-  %0 = "stablehlo.clamp"(%arg1, %arg0, %arg0) : (tensor<1xf32>, tensor<1xi32>, tensor<1xi32>) -> tensor<1xi32>
+func.func @clamp_c1(%arg0: tensor<1xi32>, %arg1: tensor<2xi32>) -> tensor<1xi32> {
+  // expected-error@+1 {{min shape [2] is not scalar and is not compatible to operand shape [1]}}
+  %0 = "stablehlo.clamp"(%arg1, %arg0, %arg0) : (tensor<2xi32>, tensor<1xi32>, tensor<1xi32>) -> tensor<1xi32>
   func.return %0: tensor<1xi32>
 }
 
 // -----
 
-func.func @clamp_invalid_clamp_min_shape(%arg0: tensor<1xi32>, %arg1: tensor<2xi32>) -> tensor<1xi32> {
-  // expected-error@+1 {{min shape [2] is not scalar and is not compatible to operand shape [1]}}
-  %0 = "stablehlo.clamp"(%arg1, %arg0, %arg0) : (tensor<2xi32>, tensor<1xi32>, tensor<1xi32>) -> tensor<1xi32>
+func.func @clamp_c2(%arg0: tensor<1xi32>, %arg1: tensor<2xi32>) -> tensor<1xi32> {
+  // expected-error@+1 {{max shape [2] is not scalar and is not compatible to operand shape [1]}}
+  %0 = "stablehlo.clamp"(%arg0, %arg0, %arg1) : (tensor<1xi32>, tensor<1xi32>, tensor<2xi32>) -> tensor<1xi32>
   func.return %0: tensor<1xi32>
 }
 
 // -----
 
-func.func @clamp_invalid_clamp_max_shape(%arg0: tensor<1xi32>, %arg1: tensor<2xi32>) -> tensor<1xi32> {
-  // expected-error@+1 {{max shape [2] is not scalar and is not compatible to operand shape [1]}}
-  %0 = "stablehlo.clamp"(%arg0, %arg0, %arg1) : (tensor<1xi32>, tensor<1xi32>, tensor<2xi32>) -> tensor<1xi32>
+func.func @clamp_c3(%arg0: tensor<1xi32>, %arg1: tensor<1xf32>) -> tensor<1xi32> {
```

Comment:
This test is covered by the ODS.

Reply:
Yes, our test guideline is that for verification in ODS, the tests are optional. So I keep it as is (just update its name)

Reply:
I prefer to remove it since keeping it would mean we have to maintain the tests covered by the ODS. Also, the [TypeInference guideline P2](https://github.com/openxla/stablehlo/blob/main/docs/type_inference.md#p2-make-the-most-of-ods) verification code related to ODS should be removed. One issue I encountered in PadOp #1005 was that it did not have "AllTypesMatch" Trait. Adding this meant removing verification tests, and also modifying the error messages of the ODS tests. I chose to completely remove ODS tests since it is redundant.

We can keep the discussion open for the next reviewer.

Reply:
Thanks guys! Happy to see we have an agreement: Remove the clamp_c3 in next commit.